### PR TITLE
New module: manage Amazon CloudFront origin access identities (cloud/amazon/cloudfront_origin_access_identity)

### PR DESCRIPTION
**Pull request by wilvk**
##### SUMMARY

As advised by @ryansb in PR #24292 this comprises the origin access identity component of the cloudfront module additions.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

##### ANSIBLE VERSION

```ansible 2.4.0 (cloudfront_invalidation 3343a2f160) last updated 2017/05/13 08:00:13 (GMT +1100)
  config file =
  configured module search path = [u'/Users/willvk/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/willvk/Source/gits/ansibles/ansible/lib/ansible
  executable location = /Users/willvk/Source/gits/ansibles/ansible/bin/ansible
  python version = 2.7.10 (default, Feb  6 2017, 23:53:20) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]

```

##### ADDITIONAL INFORMATION

```
Migrated from #24568
```

### CODE REVIEW

**Comment by wilvk** [comment]
@s-hertel have migrated this PR from https://github.com/ansible/ansible/pull/24568

File: lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+description:
+    - Allows for easy creation, updating and deletion of origin access
+      identities.
+
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+
+version_added: "2.4"
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py]: `version_added: "2.5"`

File: lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+author: Willem van Ketwich (@wilvk)
+
+options:
+    state:
+      description:
+        - The state of the resource. Valid states are
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py]:
```
description:: If the named resource should exist.
choices:
  - present
  - absent
default: present
```

File: lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        - A unique identifier to reference the origin access identity by.
+      required: false
+
+'''
+
+EXAMPLES = '''
+
+# create an origin access identity
+
+- cloudfront_origin_access_identity:
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py]: Please use `- name: ` and not comments, e.g.

```
- name: Create an origin access identity
  cloudfront_origin_access_identity:
...
```

File: lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        origin_access_identity_id=dict(),
+        caller_reference=dict(),
+        comment=dict(),
+    ))
+
+    result = {}
+    e_tag = None
+    changed = False
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=False)
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py:251]: If checkmode isn't supported please add this in the 
`notes:` section in `DOCUMENTATION`

File: lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py (hunk 2 above)
Comment (wilvk) [reply lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py]: done

File: lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py (hunk 4 above)
Comment (wilvk) [reply lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py:251]: have added to documentation under notes: section. let me know if this is incorrect

Comment (gundalow) [reply lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py:251]: Looks good, thanks

Comment (gundalow) [reply lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py:251]: oh, appologies, I think this needs to be a list

**Review by ryansb — APPROVED** [review]

**Review by s-hertel — APPROVED** [review]

**Comment by wilvk** [comment]
@ryansb @s-hertel @gundalow thanks for your help with this PR. Looks like the tests are passing now.

**Comment by s-hertel** [comment]
Thanks @wilvk for all the time you've spent on these modules!

**Pull request merged**